pkg/provisioner/controller.go

```diff
@@ -291,6 +293,103 @@ func (c *obcController) handleProvisionClaim(key string, obc *v1alpha1.ObjectBuc
 	// to control access to static buckets via RBAC rules on storage classes.
 	isDynamicProvisioning := isNewBucketByStorageClass(class)
 
+	// Re-Get the claim in order to shorten the race condition where the claim was deleted after provisioning started
+	obc, err = claimForKey(key, c.libClientset)
+	if err != nil {
+		if errors.IsNotFound(err) {
+			return fmt.Errorf("OBC was lost before we could provision: %v", err)
+		}
+		return err
+	}
+
+	// Older versions of this library set obc.Spec.BucketName with the name of the bucket generated
+	// from obc.Spec.GenerateBucketName, which causes an error condition when updating to a newer
+	// version of this library if provisioning runs again. If the bucket name has the
+	// GenerateBucketName prefix, we can assume that this was originally intended to be a generated
+	// name.
+	if obc.Spec.GenerateBucketName != "" && obc.Spec.BucketName != "" {
+		if strings.HasPrefix(obc.Spec.BucketName, obc.Spec.GenerateBucketName+"-") {
```

Comment:
I think the generation of a random bucket name should be made a helper func. I don't like that the algorithm for a creating a bucket name is found in 2 places in the code and thus can get out of sync.